src/weather.cpp

```diff
@@ -73,8 +74,15 @@ int get_hourly_rotpoints_at_temp( int temp );
 
 time_duration get_rot_since( const time_point &start, const time_point &end, const tripoint &location )
 {
-
     time_duration ret = 0;
+    // root cellars are considered as underground storage, ignores weather
+    if ( g->m.tername( location ) == "root cellar" ) {
```

Comment:
Doesn't work anyway in my test compilation, nor for @Night-Pryanik 's idea above.
Currently testing it to its limits to find what's going on.
I might have a lead...

The front of my test function now looks like this:

`    ter_id terrain = g->m.ter( location );
    add_msg( m_info, "Debug TP X: %s", location.x);
    add_msg( m_info, "Debug TP Y: %s", location.y);
    add_msg( m_info, "Debug TP Z: %s", location.z);
    add_msg( m_info, "Debug MS: %s", g->m.getmapsize() );
    if ( terrain == t_rootcellar ) {
        add_msg( "Debug 1: PING" );
        add_msg( m_info, "Debug 1: T: %s", terrain.id().c_str() );
        add_msg( m_info, "Debug 1: RC: %s", t_rootcellar.id().c_str() );
        for( time_point i = start; i < end; i += 1_hours ) {
        ret += std::min( 1_hours, end - i ) / 1_hours * get_hourly_rotpoints_at_temp( AVERAGE_ANNUAL_TEMPERATURE ) * 1_turns;
        } 
        return ret;
    } else { 
        add_msg( "Debug 1: PONG" );
        add_msg( m_info, "Debug 2: T: %s", terrain.id().c_str() );
        add_msg( m_info, "Debug 2: RC: %s", t_rootcellar.id().c_str() );
    }
    if ( terrain == t_pit ) { add_msg( "Debug 2: PIT PIT PIT" ); }`

RESULT: 
X = 678, Y = 1983, Z = 0
MS=11 (?)
Debug 2: T: t_null
Debug 2: RC: t_rootcellar

t-null --> this suggests that map::ter() returns t_null, and that might be becouse XYZ are !inbound(), but why? maybe the strange getmapsize() result?

Reply:
The huge coords mean that you're getting either absolute coords or coords in overmap. But `map::ter` expects local coords, which are not always available here (rot processing can happen when local map doesn't even exist).
This means that the approach with root cellar detection at rot calculation level can't work reliably.

Reply:
I found the answer: 'tripoint location' argument pushed to the function shuld be first converted by mapp:getlocal() to local coordinates. I knew it, I knew it must have been coordinates conversion

Reply:
> This is called often - at least once per item. This check may be expensive since it affects all such calls.

How to avoid it then? I think it has to check where the item is located. Also for the sake of futer PRs, if I follow Kevin's idea on storing food in proper furniture, I would use this method to discern where is it located, and if such location is valid.

Reply:
You'd need to pass it from above, where item location in local coords is still available.
If you do it this way, make sure all paths go through the same functions, to avoid hard to find bugs where rot is different depending on how is it processed.